app/web/src/components/EditorMenu/AllShortcutsModal.tsx

```diff
@@ -0,0 +1,44 @@
+import { Menu } from '@headlessui/react'
+import { useEffect, useState } from 'react'
+import { useHotkeys } from 'react-hotkeys-hook';
+import { makeStyles } from '@material-ui/core/styles'
+import Dialog from '@material-ui/core/Dialog'
+import { editorMenuConfig } from './menuConfig';
+import { useIdeContext } from 'src/helpers/hooks/useIdeContext'
+
+const SHORTCUT = 'ctrl+/, command+/'
```

Comment:
Should this also have a button in the menu, how will users learn how to get this modal up?

Why did you choose this combo, is this a pattern elsewhere? I'm concerned that is the same hotkey as commenting a line in the editor, since the editor is currently swallowing this hotkey so it still works in the editor but seems weird having two different behaviour depending on where you're focused.

Maybe it's a mac only thing, but similar to the above cmd alt R only resets the viewer when I'm not focused on the edit, because its already a hotkey in the editor. Seems like we're going to be competing for hotkeys from the monaco editor.

Reply:
Ah crap this is all good consideration @Irev-Dev.

Here are my recommendations for these 2:
- **All Shortcuts**:`command+/` 👉🏻  `ctrl+shift+?`. This is what Figma's shortcut to show shortcuts is. I agree this should have a button in the menu. Would it go under View?
- **Reset Layout**: `command+alt+R` 👉🏻  `ctrl+shift+R`. It appears that `ctrl+shift` is a fairly reliable key combination.

Separately, should we have a part of this component or its tests have a list of the Monaco shortcuts to verify any new ones aren't stepping on them?

Reply:
One thing that's weird to me is that HotKeys treats `Ctrl` and `Cmd` as the Windows/Mac alternatives, but it's really `Cmd` and `Win` keys that are analogous, usually called the OS key within the browser, and `Ctrl` is a key that both OSes have.

Reply:
- ctrl shift ? I think make sense for the modal :)
- test might be good, but not sure if there is a way it pull hotkeys out of monaco without hardcoding them ourselves.
- I think the library does that for practical reasons since ctrl is the most common key used for hotkey for win and cmd is the most common for mac, for example cmd c, cmd v is copy paste.

Reply:
Yeah doesn't appear to be a great way to pull them out programmatically, only was able to find [this GH issue comment about it](https://github.com/microsoft/monaco-editor/issues/823#issuecomment-470754000). There doesn't appear to be like `KeybindingsRegistry.getAll()` method or anything like that.

The Monaca online editor is built on top of Monaco and has this hardcoded [list of shortcuts](https://docs.monaca.io/en/products_guide/monaca_ide/editor/) that we could pull from for a hardcoded solution. I think they'll be pretty stable so that might not be too bad? And [this issue has been raised](https://github.com/microsoft/monaco-editor/issues/102) in Monaco for an easy API for swapping out keyboard shortcuts, I couldn't find a clean holistic API for keybindings at all really.

Want me to open an issue for the tests? I feel like we can do that a bit later, right?

Reply:
Yeah sure, sounds good.

Reply:
Added issue #499 to track this